presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -438,6 +432,28 @@ public void testFilesForAbortedTransactionsIgnored()
         }
     }
 
+    private void hdfsDeleteAll(String directory)
+    {
+        if (!hdfsClient.exist(directory)) {
+            return;
+        }
+        for (String file : hdfsClient.listDirectory(directory)) {
+            hdfsClient.delete(directory + "/" + file);
+        }
+    }
+
+    private void hdfsCopyAll(String source, String target)
+    {
+        if (!hdfsClient.exist(target)) {
+            hdfsClient.createDirectory(target);
```

Comment:
target.parent ?

Reply:
Not needed for current usecases. As this is local helper function I do not want to bloat it too much. If we decide to make a general helper util from it it would need to be smareter.
But then probably it should just be extension of `HdfsClient` interface.

Reply:
i misread the code